### ISSUE TITLE
Rocket.Chat won't start - cannot perform operation: mount --bin /snap/core/current/etc/alteratives ...

### ISSUE THREAD

**Issue by tamasgal** [RocketChat/Rocket.Chat#12502]
Today I did some `apt-get upgrades` on our main chat server and afterwards noticed that the chat is not responding anymore. I restarted the container (that usually fixed such problems) but no luck.

I get these log outputs with `journalctl`:

```
# journalctl -f -u snap.rocketchat-server.rocketchat-server
-- Logs begin at Wed 2018-10-31 16:41:46 CET. --
Oct 31 16:41:49 pi1145 systemd[1]: Starting Service for snap application rocketchat-server.rocketchat-server...
Oct 31 16:41:49 pi1145 systemd[1]: Started Service for snap application rocketchat-server.rocketchat-server.
Oct 31 16:41:49 pi1145 rocketchat-server.rocketchat-server[628]: cannot perform operation: mount --bind /snap/core/current//etc/alternatives /tmp/snap.rootfs_NVl2Ai/etc/alternatives: No such file or directory
Oct 31 16:41:49 pi1145 systemd[1]: snap.rocketchat-server.rocketchat-server.service: main process exited, code=exited, status=1/FAILURE
Oct 31 16:41:49 pi1145 systemd[1]: Unit snap.rocketchat-server.rocketchat-server.service entered failed state.
Oct 31 16:41:49 pi1145 systemd[1]: snap.rocketchat-server.rocketchat-server.service holdoff time over, scheduling restart.
Oct 31 16:41:49 pi1145 systemd[1]: Stopping Service for snap application rocketchat-server.rocketchat-server...
Oct 31 16:41:49 pi1145 systemd[1]: Starting Service for snap application rocketchat-server.rocketchat-server...
Oct 31 16:41:49 pi1145 systemd[1]: snap.rocketchat-server.rocketchat-server.service start request repeated too quickly, refusing to start.
Oct 31 16:41:49 pi1145 systemd[1]: Failed to start Service for snap application rocketchat-server.rocketchat-server.
```

I am wondering why that file is suddenly gone? I have not much experiences with snaps, but I tried to revert to version 0.69.2 and also removed and installed it again. I now have 0.70.4 installed.

This is the output of the `systemctl status ...caddy...`:

```
# systemctl status snap.rocketchat-server.rocketchat-caddy.service
* snap.rocketchat-server.rocketchat-caddy.service - Service for snap application rocketchat-server.rocketchat-caddy
   Loaded: loaded (/etc/systemd/system/snap.rocketchat-server.rocketchat-caddy.service; enabled)
   Active: failed (Result: start-limit) since Wed 2018-10-31 16:39:53 CET; 9s ago
  Process: 1255 ExecStart=/usr/bin/snap run rocketchat-server.rocketchat-caddy (code=exited, status=1/FAILURE)
 Main PID: 1255 (code=exited, status=1/FAILURE)

Oct 31 16:39:53 pi1145 systemd[1]: Unit snap.rocketchat-server.rocketchat-caddy.service entered failed state.
Oct 31 16:39:53 pi1145 systemd[1]: snap.rocketchat-server.rocketchat-caddy.service holdoff time over, scheduling restart.
Oct 31 16:39:53 pi1145 systemd[1]: Stopping Service for snap application rocketchat-server.rocketchat-caddy...
Oct 31 16:39:53 pi1145 systemd[1]: Starting Service for snap application rocketchat-server.rocketchat-caddy...
Oct 31 16:39:53 pi1145 systemd[1]: snap.rocketchat-server.rocketchat-caddy.service start request repeated too quickly, refusing to start.
Oct 31 16:39:53 pi1145 systemd[1]: Failed to start Service for snap application rocketchat-server.rocketchat-caddy.
Oct 31 16:39:53 pi1145 systemd[1]: Unit snap.rocketchat-server.rocketchat-caddy.service entered failed state.
```

Any ideas?

**Comment by tamasgal** [comment]
Just another note, I tried to call `rocketchat-server.initcaddy` and got the same errors:

```
root@pi1145:~# rocketchat-server.initcaddy
cannot perform operation: mount --bind /snap/core/current//etc/alternatives /tmp/snap.rootfs_814VQ2/etc/alternatives: No such file or directory
root@pi1145:~# snap run rocketchat-server.initcaddy
cannot perform operation: mount --bind /snap/core/current//etc/alternatives /tmp/snap.rootfs_O63fEC/etc/alternatives: No such file or directory
```

**Comment by reetp** [comment]
You haven't used a Issue template to fully describe your system.


 mentioned stuff you put on the open.rocket.chat

I have nor real idea about snaps but the bit that stands out here is

mount --bind /snap/core/current//etc/alternatives /tmp/snap.rootfs_814VQ2/etc/alternatives: No such file or directory

Why is it using /tmp ? That directory probably gets cleared on a reboot (because it is a temporary location)

Where do snaps store their data ? https://askubuntu.com/questions/762354/where-can-ubuntu-snaps-write-data#762405

You might want to try running up a snap on a test VM and see what it does and where data is stored.

**Comment by tamasgal** [comment]
I have not used the template since I had the feeling it's a snap issue.

I am currently trying to figure out how to get the data out of the mongodb but it seems to be quite hard when the mongodb process won't start up :(

**Comment by reetp** [comment]
Always use the template to give proper information. It can be important.

As a follow up....

OP runs Xen containers. Install was into a Debian Jessie container using snap from backports (risky business)

On starting the snap will not run - logs above. As a result he can't get into Rocket to manually backup Mongo, and can't see to copy the install to a new location either.

There was a suggestion in the #ubuntu-snap chat 

"I have installed the snap whith a new rocket version. But, how can I access the mongodb? I need migrate from old mongodb to new server with snap/new version.."

"you can access like this `sudo rocketchat-server.mongo` "

**Comment by tamasgal** [comment]
Many thanks for the follow @reetp 

The problem is, that I cannot start anything which is related to snap:

```
root@pi1145:/# rocketchat-server.mongo
cannot perform operation: mount --bind /snap/core/current//etc/alternatives /tmp/snap.rootfs_Wukvjg/etc/alternatives: No such file or directory
```

**Comment by tamasgal** [comment]
Alright, I got it back to live and everything seems to work. I still cannot believe it.

The procedure was the following:

- install a fresh new Xen container (on fresh virtual disks) with debian jessie
- update all the packages to the latest versions
- do a full dist upgrade to debian stretch
- install snap with `apt` (and other related packages like apache2, letsencrypt...)
- copy over /var/snap/rocketchat-server/common from the system backup to the new server (same location)
- copy over individual configuration files (apache reverse proxy configuration, letsencrypt, etc.)
- `snap install rocketchat.server`
- everything works. reboot to make sure it comes back and enjoy

All in all the solution is quite straight forward and I almost regret that I spent so much time and effort to make this work under the same system (Debian Jessie). I guess the biggest show blocker was the fact that at some point they removed `snapd` from the jessie-backports (which I learned the hard way when I tried to reinstall it), so I simply wasn't able to restore a working snap installation. But whatever...
